### DISCUSSION TITLE
External redis and sql container possible?

### DISCUSSION THREAD

**Discussion by 3x3cut0r** [AzuraCast/AzuraCast#5119] [General]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Stable Channel

### What happened?

I have a docker rootless installation on a vps.
I have some abnormal pain  while deploying docker containers with some provided scripts like in this case.
While running hundreds of containers i need to follow some IP and naming conventions which i absolutely have control of.

So is it possible to setup redis and the db in external networks with some custom names?
For example my redis container in not named "redis" ... which runs into some errors (see log)

Where can i specify this one?
I didn't find anything in the documentation.

Same with the db

### Relevant log output

```shell
Warning: Redis::connect(): php_network_getaddresses: getaddrinfo for redis failed: Name or service not known in /var/azuracast/www/bin/uptime_wait on line 134
Call Stack:
    0.0008     417880   1. {main}() /var/azuracast/www/bin/uptime_wait:0
    0.0011     422312   2. UptimeWait->run() /var/azuracast/www/bin/uptime_wait:167
   62.4371     422600   3. UptimeWait->checkRedis() /var/azuracast/www/bin/uptime_wait:77
   62.4371     422648   4. Redis->connect($host = 'redis', $port = 6379, $timeout = 15) /var/azuracast/www/bin/uptime_wait:134
PHP Warning:  Redis::connect(): php_network_getaddresses: getaddrinfo for redis failed: Name or service not known in /var/azuracast/www/bin/uptime_wait on line 134
PHP Stack trace:
PHP   1. {main}() /var/azuracast/www/bin/uptime_wait:0
PHP   2. UptimeWait->run() /var/azuracast/www/bin/uptime_wait:167
PHP   3. UptimeWait->checkRedis() /var/azuracast/www/bin/uptime_wait:77
PHP   4. Redis->connect($host = 'redis', $port = 6379, $timeout = 15) /var/azuracast/www/bin/uptime_wait:134
```

**Comment by Vaalyn** [comment]
You can configure your own DB and Redis server in the `azuracast.env` file:
https://github.com/AzuraCast/AzuraCast/blob/main/azuracast.sample.env#L40-L102

**Comment by 3x3cut0r** [comment]
Ok thank you for that.
i now have set redis-host, port and db settings ... but redis still isn't working because it is still looking for the wrong name:

`2022/02/18 10:21:46 [error] 90#90: REDISTORE: can't initialize redis ctx 000056554BC365D8
2022/02/18 10:21:46 [alert] 54#54: worker process 90 exited with fatal code 2 and cannot be respawned
2022/02/18 10:21:46 [error] 91#91: redis_nginx_adapter: could not create the redis context for redis:6379 - Name or service not known`

redis:6379 is wrong in my case
i set it correct in the env file ... it is also showing in the docker inspect details ... so the env file is working correctly!
redis and azurecast containers are in the same docker network.
what did i wrong?

for example in my nextcloud service i did the exact same thing ... mariadb and redis are external from the nextcloud service.

thank you for some help

EDIT:
btw... i there a possibility to set a redis password in the env file?
